### PR TITLE
feat: add coffeesense-language-server support

### PR DESCRIPTION
**Pull request by micke**

### CODE REVIEW

File: lua/lspconfig/server_configurations/coffeesense.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
+local util = require 'lspconfig.util'
+
+local bin_name = 'coffeesense-language-server'
+local cmd = { bin_name, "--stdio" }
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/coffeesense.lua:4]:
```suggestion
local cmd = { bin_name, '--stdio' }
```

File: lua/lspconfig/server_configurations/coffeesense.lua [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+if vim.fn.has 'win32' == 1 then
+  cmd = { 'cmd.exe', '/C', bin_name, "--stdio" }
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/coffeesense.lua:7]:
```suggestion
  cmd = { 'cmd.exe', '/C', bin_name, '--stdio' }
```

File: lua/lspconfig/server_configurations/coffeesense.lua [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+end
+
+return {
+  default_config = {
+    cmd = cmd,
+    filetypes = { 'coffee' },
+    root_dir = util.root_pattern('package.json'),
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/coffeesense.lua:14]:
```suggestion
    root_dir = util.root_pattern 'package.json',
```

**Comment by micke** [comment]
@glepnir Thanks, sorry about the missed strings, that was sloppy of me not to check the CI...

**Comment by glepnir** [comment]
It's ok :)

**Pull request merged**